src/input_context/bind.rs

```diff
@@ -0,0 +1,172 @@
+use crate::prelude::*;
+
+#[derive(Debug, Default)]
+pub struct BindConfigInfo {
+    pub modifiers: Vec<Box<dyn InputModifier>>,
+    pub conditions: Vec<Box<dyn InputCondition>>,
+}
```

Comment:
Maybe inline this two? Yes, it will require adding two more methods for getting it, but calling `self.conditions()` is nicer then `self.info().conditions`.

Reply:
agreed, inlined into the structs themselves